doc/md/versioned/06-custom-migrations.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Custom migrations
+id: custom-migrations
+---
+## Supporting repository
+
+The change described in this section can be found in
+[PR #7](https://github.com/rotemtam/ent-versioned-migrations-demo/pull/7/files)
+in the supporting repository.
+
+## Custom migrations
+In some cases, you may want to write custom migrations that are not automatically
+generated by Atlas. This can be useful in cases where you want to perform changes
+to your database that aren't currently supported by Ent, or if you want to seed
+the database with data. 
+
+In this section, we will learn how to add custom migrations to our project. For the
+purpose of this guide, let's assume we want to seed the users table with some data.
+
+## Create a custom migration
+
+Let's start by adding a new migration file to our project:
+
+```shell
+atlas migrate new seed_users --dir file://ent/migrate/migrations
+```
+
+Observe that a new file named `20221115102552_seed_users.sql` was created in the
+`ent/migrate/migrations` directory. 
+
+Continue by opening the file and adding the following SQL statements:
+
+```sql
+INSERT INTO `users` (`name`, `email`, `title`)
+VALUES ('Jerry Seinfeld', 'jerry@seinfeld.io', 'Mr.'),
+       ('George Costanza', 'george@costanza.io', 'Mr.')
+```
+
+## Recalculating the checksum file
+
+Let's try to run our new custom migration:
+
+```shell
+atlas migrate apply --dir file://ent/migrate/migrations --url mysql://root:pass@localhost:3306/db
+```
+Atlas fails with an error:
+```text
+You have a checksum error in your migration directory.
+This happens if you manually create or edit a migration file.
+Please check your migration files and run
+
+'atlas migrate hash'
+
+to re-hash the contents and resolve the error
+
+Error: checksum mismatch
+```
+Atlas introduces the concept of [migration directory integrity](/concepts/migration-directory-integrity)
```

Comment:
Link to Atlas instead: https://atlasgo.io/concepts/migration-directory-integrity